static/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -238,6 +255,10 @@ class SpanTree extends Component<PropType> {
             />
           );
           acc.spanNumber = spanNumber + 1;
+
+          acc.numOfSpansOutOfViewAbove = 0;
+          acc.filteredSpansAbove = [];
```

Comment:
This was what was causing the bug related to hidden span counts not being correct; we need to reset the span counts/array when pushing an autogroup component, or the count will carry over with the next iteration